core/lib/vlog/src/lib.rs

```diff
@@ -301,6 +312,8 @@ impl ObservabilityBuilder {
             None
         };
 
+        // We use backtrace capture in `json_panic_handler` so we have to have it always enabled
+        std::env::set_var("RUST_BACKTRACE", "1");
```

Comment:
We only use `json_panic_handler` for JSON log format, no?

Reply:
Also, this overrides `RUST_BACKTRACE=full` which is probably an undesired side-effect

Reply:
everything non zero is the same 
https://doc.rust-lang.org/std/backtrace/#environment-variables.

We use json logs as well

Reply:
Ah, apparently `full` is not a thing anymore indeed

> We use json logs as well

Yeah I just meant that `json_panic_handler` is not called 100% of the time we run our code, although I assume all real environments use JSON logs